### DISCUSSION TITLE
Can HTML be used inside Markdown?

### DISCUSSION THREAD

**Discussion by roke-julian-lockhart** [mermaid-js/mermaid#5329] [Q&A] [answered]
Using https://marketplace.visualstudio.com/_apis/public/gallery/publishers/bierner/vsextensions/markdown-mermaid/1.22.0/vspackage (from https://marketplace.visualstudio.com/items?itemName=bierner.markdown-mermaid) it appears that every HTML element I try to include in my Markdown strings (<code>`"</code>) are reported as unsupported. Are any supported, and are they documented?

**Comment by ajmalfaris11** [comment, marked as answer]
In Markdown, the syntax for HTML elements is supported, but it depends on the specific Markdown renderer or parser being used. Some Markdown parsers may have limitations or restrictions on which HTML elements are allowed for security or other reasons.

Here's how you can include HTML elements in Markdown:

1. **Inline HTML**: You can include HTML elements directly within Markdown content. For example:

   ```markdown
   This is a paragraph with <strong>bold</strong> text.
   ```

2. **Block-Level HTML**: You can also use HTML block-level elements like `<div>`, `<p>`, `<ul>`, `<ol>`, etc., which will be rendered as block-level elements in HTML.

   ```markdown
   <div>
       <p>This is a paragraph.</p>
       <ul>
           <li>Item 1</li>
           <li>Item 2</li>
       </ul>
   </div>
   ```

3. **HTML Attributes**: You can use HTML attributes with Markdown elements. For example:

   ```markdown
   <img src="image.jpg" alt="Image">
   ```

However, it's essential to note that not all HTML elements may be supported or rendered correctly by all Markdown parsers. Additionally, some Markdown parsers may have security restrictions or limitations on which HTML elements are allowed to prevent XSS (Cross-Site Scripting) attacks.

If you're encountering issues with including HTML elements in your Markdown content, you may need to consult the documentation or specifications of the Markdown parser you're using to understand its capabilities and any restrictions it may have regarding HTML elements. Additionally, testing your Markdown content with different parsers or rendering environments can help identify compatibility issues.

**Reply by roke-julian-lockhart** [reply]
> https://github.com/orgs/mermaid-js/discussions/5329#discussioncomment-8646264

Thanks, @ajmalfaris11. I've filed https://github.com/mjbvz/vscode-markdown-mermaid/issues/246#issue-2163953175 and https://github.com/mermaid-js/mermaid-live-editor/issues/1411#issue-2163963447 consequently. Is HTML support in Markdown strings documented as expected functionality?
